File: Data Structures & Algorithms/max-water-container/submission-5.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int currMax = 0;
        int l = 0;
        int r = heights.size() - 1;

        while (l < r) {
            int dist = r - l;

            if (heights[l] < heights[r]) {
                currMax = max(currMax, (heights[l] * dist));
                l++;
                continue;

            } else if (heights[r] < heights[l]) {
                currMax = max(currMax, (heights[r] * dist));
                r--;
                continue;

            } else {
                currMax = max(currMax, (heights[l] * dist));
                l++;
            }
        }

        return currMax;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        currMax = 0

        for r in range(1, len(prices)):
            total = prices[r] - prices[l]

            if total <= 0:
                l = r
            else:
                currMax = max(currMax, total)

        return currMax